bn_mp_set_double.c

```diff
@@ -11,8 +11,21 @@
  *
  * SPDX-License-Identifier: Unlicense
  */
+#include <float.h>
+/* Minimum information needed */
+#if ( (defined DBL_MAX) && (FLT_RADIX == 2) )
 
-#if defined(__STDC_IEC_559__) || defined(__GCC_IEC_559)
+/* We can use a faster method if we have an IEEE compliant machine and a working stdint.h */
+#if (((defined __STDC_IEC_559__) || (defined __GCC_IEC_559)) && (defined UINT64_MAX))
```

Comment:
Check for MP_HAS_NICE_FLOATING_POINT instead and move this #if to the header where you do the check for apple.

Reply:
Can do the first, but the check for `uint64_t` needs to stay because of the "without stdint.h" folks. I think I have mentioned them?